### PR TITLE
Customize search component and blog results

### PR DESCRIPTION
**Pull request by samkim**
Copied search components from Nextra and customized search to include blog post content in results.

### CODE REVIEW

**Comment by jzelinskie** [comment]
LGTM from a usage perspective -- tagging @josephschorr for the JS review

File: components/nextra/types.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Forked from https://github.com/shuding/nextra/blob/2e78fe5f52a523399eb491fe525b67c7534f2f0e/packages/nextra-theme-docs/src/types.ts
```
Comment (josephschorr) [inline components/nextra/types.ts:1]: Include the nextra license somewhere

Comment (samkim) [reply components/nextra/types.ts:1]: Copied into forked files

**Review by josephschorr — APPROVED** [review]
LGTM

**Pull request merged**